### DISCUSSION TITLE
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)

### DISCUSSION THREAD

**Discussion by yonglie** [milvus-io/milvus#15763] [Q&A and General discussion]
我已经用了gpu的milvus，算力6.1，但是还是报错了  

 __  _________ _   ____  ______
   /  |/  /  _/ /| | / / / / / __/
  / /|_/ // // /_| |/ / /_/ /\ \
 /_/  /_/___/____/___/\____/___/

Welcome to use Milvus!
Milvus Release version: v1.0.0, built at 2021-03-04 06:42.57, with OpenBLAS library.
You are using Milvus GPU edition
Last commit id: 21ea92ec7e57ab00cca671d74195c4eaa81869a2

Loading configuration from: /var/lib/milvus/conf/server_config.yaml
GPU resources ENABLED !
Supported CPU instruction sets: avx2, sse4_2
Nvidia driver version: 510.47.03
CUDA Driver Version / Runtime Version : 11.6 / 10.1
FAISS hook AVX2
Milvus server started successfully!
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 0 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 819200000 B, highwater 134217728 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 819200000 B, highwater 819200000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 262144 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 131072 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 131072 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 131072 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 8388608 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 8388608 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 2048 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 2048 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 2048 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 2048 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 268435456 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 15184 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 15184 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 30368 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 30368 B, highwater 1638400000 B)
WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 131072 B, highwater 1638400000 B)
Faiss assertion 'err == cudaSuccess' failed in void faiss::gpu::freeMemorySpace(faiss::gpu::MemorySpace, void*) at gpu/utils/MemorySpace.cpp:72; details: Failed to cudaFree pointer 0x7f340c000000 (error 1 invalid argument)

**Comment by yhmo** [comment]
The CUDA api `cudaFree()` throws "invalid argument" error, not sure why.
Firstly use `nvidia-smi` to print the GPU informations.

**Comment by yhmo** [comment]
Could you show me your server_config.yaml, how you config the "gpu" section?

**Comment by yonglie** [comment]
谢谢您，我现在把相似度的类型改为L2了，但是不知道如何计算欧氏距离，
得到了
List<List<Float&gt;&gt; IP = sr.getResultDistancesList();
一个这样的ResultDistancesList
,以前我用IP方式的时候，可以用这个公式得到：euc = Math.sqrt(2*(1−cos(A,B)))


现在不知道sr.getResultDistancesList();得到的什么了？




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "milvus-io/milvus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年2月28日(星期一) 下午5:30
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [milvus-io/milvus] WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B) (Discussion #15763)





 
Could you show me your server_config.yaml, how you config the "GPU" section?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by yhmo** [reply]
search时可以传入多条向量（用nq表示），每条向量计算得到topk个距离，所以search返回的结果是一个二维数组
List<List<Long>> getResultIdsList()
List<List<Float>> getResultDistancesList()

List<List<Long>> 指的是n条向量按顺序计算得到的n组topk个id
List<List<Float>> 指的是n条向量按顺序计算得到的n组topk个距离，这里的距离是欧式距离的平方（比较距离远近不需要开方）

**Comment by yonglie** [comment]
我现在用2.0.1的milvus版本，发现一开始搜索还准确，后面就不准确了，最小的score都23左右，怎么会这样呢？


而且有报错信息如下：


Unable to add the resource at [/f1/56ae934f-638e-4ac4-adb3-bfba9a173040.jpg] to the cache for web application [/uploadmilvus] because there was insufficient free space available after evicting expired cache entries - consider increasing the maximum size of the cache








------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "milvus-io/milvus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年2月28日(星期一) 晚上6:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [milvus-io/milvus] WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B) (Discussion #15763)





 
search时可以传入多条向量（用nq表示），每条向量计算得到topk个距离，所以search返回的结果是一个二维数组
 List<List&gt; getResultIdsList()
 List<List&gt; getResultDistancesList()
 
List<List&gt; 指的是n条向量按顺序计算得到的topk个id
 List<List&gt; 指的是n条向量按顺序计算得到的topk个距离，这里的距离是欧式距离的平方（比较距离远近不需要开方）
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by yhmo** [reply]
distance是每个维度相减的平方和，两条向量如果维度较高，只要有少许不同，得到的值就很大。
“一开始还准确，后面就不准确”，这里“一开始”和“后面”之间做了什么操作？没有具体的描述就无法判断出了什么问题。

**Comment by yonglie** [comment]
再问问您，我用的是java的api，我loadcollection的时候，就卡住了
milvusClient.loadCollection(LoadCollectionParam.newBuilder()
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; .withCollectionName(collectionName)
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; .build());



然后那个工具也是卡住了

Attu





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "milvus-io/milvus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月2日(星期三) 下午3:26
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [milvus-io/milvus] WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B) (Discussion #15763)





 
distance是每个维度相减的平方和，两条向量如果维度较高，只要有少许不同，得到的值就很大。
 “一开始还准确，后面就不准确”，这里“一开始”和“后面”之间做了什么操作？没有具体的描述就无法判断出了什么问题。
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by yhmo** [reply]
之前在2.0.0之前的预览版本有比较多这种问题，但在2.0.1里应该基本杜绝了。如果你用的是standalone，就把milvus容器的日志输出来给我们看一下。

**Comment by yonglie** [comment]
这个就是我跑代码的时候打印的，在load的时候卡住了，附件里是我操作的视频


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "milvus-io/milvus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月2日(星期三) 晚上6:23
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [milvus-io/milvus] WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B) (Discussion #15763)





 
之前在2.0.0之前的预览版本有比较多这种问题，但在2.0.1里应该基本杜绝了。如果你用的是standalone，就把milvus容器的日志输出来给我们看一下。
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android. 
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;


从QQ邮箱发来的超大附件

Video_2022-03-02_182723.wmv (9.69M, 无限期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?k=633564378dd45d9baec2020b40320b4b04535c56560200051e025751001f0d560704490e02010f49505657010004015d0003520e662939325a51015839000956011854044b020b3b020d5600540117135e43640a&t=exs_ftn_download&code=35d7f29d

**Comment by yonglie** [comment]
这些是那些容器的日志。


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "milvus-io/milvus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月2日(星期三) 晚上6:23
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [milvus-io/milvus] WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 134217728 B, highwater 134217728 B) (Discussion #15763)





 
之前在2.0.0之前的预览版本有比较多这种问题，但在2.0.1里应该基本杜绝了。如果你用的是standalone，就把milvus容器的日志输出来给我们看一下。
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by yhmo** [reply]
没看到有下载链接。你还是去issues里面开一个issue吧，把日志贴在issue里面，会有人处理的。

**Comment by yonglie** [comment]
服务器配置：
ubuntu18.04
251G内存
7.3TB固态硬盘没有显卡
一共2个cpu
单个cpu的物理核数：12
总逻辑核数：48

docker,milvus,mysql,都在固态硬盘中运行。数据也都存储在固态硬盘中。

头像库：1586454条数据
产品库：5550835条数据

头像库搜索：耗时7秒
产品库搜索：耗时10秒

还有哪些因素可以直接明显的提高搜索速度呢？
